.github/ISSUE_TEMPLATE/bug_report.md

```diff
@@ -10,23 +10,46 @@ assignees: ''
 
 The release number or commit hash of the version you're using.
 
-## Operating system & version
+## \[Optional\] Relevant system infromation
 
-Ex: Windows 10, Ubuntu 18.04, iOS 14.
+If you cannot get Bevy to build or run on your machine, please include:
 
-## What you did
+- the Rust version you're using (you can get this by running `cargo --version`)
+  - Bevy relies on the "latest stable release" of Rust
+  - nightly should generally work, but there are sometimes regressions: please let us know!
+- the operating system or browser used, including its version
+  - e.g. Windows 10, Ubuntu 18.04, iOS 14
+
+If your bug is rendering-related, copy the adapter info that appears when you run Bevy.
 
-The steps you took to uncover this bug. Please list full reproduction steps if
-feasible.
+```ignore
+`AdapterInfo { name: "NVIDIA GeForce RTX 2070", vendor: 4318, device: 7938, device_type: DiscreteGpu, backend: Vulkan }`
+```
 
-## What you expected to happen
+You should also consider testing the examples of our upstream dependencies to help isolate any setup-specific issue:
 
-What you think should've happened if everything was working properly.
+- [`wgpu`](https://github.com/gfx-rs/wgpu) for rendering problems
+- [`winit`](https://github.com/rust-windowing/winit) for input and window management
+- [`gilrs`](https://docs.rs/gilrs/latest/gilrs/) for gamepad inputs
+
+## What you did
 
-## What actually happened
+Describe how you arrived at the problem. If you can, consider providing a code snippet or link.
```

Comment:
If users give links to reproductions, they often choose to keep them all rights reserved (e.g. https://github.com/bevyengine/bevy/issues/4793).
This hampers the usability of these reproductions significantly, as e.g. they cannot be used as the basis of regression tests, among many other things. Afaik, there isn't even a license to download the code.
We should ask users to ensure that any reproductions are in the form of code snippets (or are compatibly licensed, though I find it unlikely that users would correctly compatibly license their code)

Reply:
I don't think that's a conscious decision, more often than not it's people not bothering to add a license or simply beginners not aware of the potential issues it may cause.

As for downloading it, when creating a github you agreed to let other github users download and view your code.

> If you publish your source code in a public repository on GitHub, according to the [Terms of Service](https://docs.github.com/en/free-pro-team@latest/github/site-policy/github-terms-of-service), other users of GitHub.com have the right to view and fork your repository

See the note in this page https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/licensing-a-repository

Reply:
Either way, I agree that asking for people to license their code is appropriate.

Reply:
I'm on board. Wording suggestions?

Reply:
Maybe something like that:

```suggestion
Describe how you arrived at the problem. If you can, consider providing a code snippet or link. If you share a repository, please make sure to include a compatible license such as MIT or Apache 2.0.
```

I'm not sure if we should specify a license or just let people pick one.

Reply:
For it to be useful for creating regression tests, without opening the can of worms of licensing our tests differently, it must be exactly the Apache and MIT licenses that we use.

But yes, having an MIT or Apache 2.0 license at least gives enough permission to try and understand the bug from the code. I'd rather just heavily encourage including the reproduction code inline. In almost every case where people provide a reproduction repository, the only thing anyone viewing the issue to try to reproduce will do is open `src/main.rs` (after being surprised that the repository actually has a license)

We can encourage people to use a `details` section if the reproduction code is long.

Reply:
Hey just chiming in as my issue was mentioned: The lack of license was unintentional, it just didn't cross my mind to add one!
And I should have just inlined it I think.

Reply:
Yeah, definitely. Mentioning your issue wasn't trying to call you out as such- it's a very easy thing to miss. That's why we need to make it clear here ❤️.
Indeed, your missing it was helpful, because this issue was in recent working memory for me to bring it up here :)

Indeed, now that you have a license, it's now an example of my point in the top level message of:

> I find it unlikely that users would correctly compatibly license their code

since you have created the `LICENSE.md` file, but you haven't copied the licenses from `docs/LICENSE-MIT` and `docs/LICENSE-APACHE`, despite referencing them in `LICENSE.md`. 
This isn't very important - it's obvious to any observer that you're intending for the code to be licensed as MIT/Apache 2.0, which is *probably* good enough. But it's much easier for us to be confident if it's just included as code snippets in the issue.

Reply:
Oh right! Well I'm glad to serve as a test subject for this discussion haha

Reply:
Side note on inlining: It find github's markdown very unintuitive. The text input box allows you to paste tabs, but for some reason the formatting will be completely broken if you use them inside a code block in a <details> section... Took me like 5 minutes to figure it out how to format it properly